enhancements/cluster-logging/gcp-external-account-auth.md

```diff
@@ -0,0 +1,201 @@
+3. Implement the WIF Authentication Flow   
+   * Step 1: Read the OpenShift service account token from the path specified in the credentials file   
+   * Step 2: Exchange the OpenShift token for a Google Identity Token via STS.   
+   * Step 3: Exchange the Identity Token for an Access Token via IAM impersonation.  
+4. Ensure Compatibility with service_account Type  
+   * The existing authentication flow for service_account remains unchanged.
+
+#### Credentials File Example
+(external_account type)
+```json
+{
+  "type": "external_account",
+  "audience": "//iam.googleapis.com/...<workload_identity_pool_provider>",
+  "token_url": "https://sts.googleapis.com/v1/token",
+  "service_account_impersonation_url": "https://iamcredentials......iam.gserviceaccount.com:generateAccessToken",
+  "credential_source": { 
+    "file": "/var/run/ocp-collector/serviceaccount/token"
+  }
+}
+
+```
+
+#### OIDC Request Examples
+
+##### How Is The Bound Token Used in GCP Workload Identity Federation (WIF)?
+1. Pod gets the Bound Service Account Token
+   * This token is created automatically by OpenShift and mapped to the service account assigned to the pod.
+2. The OpenShift Pod sends this token to GCP STS  
+   * This is Step 1 of the WIF authentication process.
+   * The token is sent to Google's Security Token Service (STS) API: https://sts.googleapis.com/v1/token
+and is included in the subject_token field of the next request.   
+3. GCP STS validates the OpenShift token and exchanges it for an OIDC token
+   * This step verifies that the OpenShift cluster is a trusted identity provider for GCP.
+   * The result is a Google Identity Token (OIDC token).
+4. The Identity Token is then exchanged for a Google Access Token using IAM impersonation
+   * The access token allows access to GCP resources.
+
+##### Example: How The Token Is Used in WIF Authentication
+When sending the OpenShift "bound" service account token to Google's STS, the payload looks like:
+```json
+{
+  "grant_type": "urn:ietf:params:oauth:grant-type:token-exchange",
+  "requested_token_type": "urn:ietf:params:oauth:token-type:access_token",
+  "subject_token_type": "urn:ietf:params:oauth:token-type:jwt",
+  "subject_token": "<OpenShift Bound Service Account Token>",
+  "audience": "//iam.googleapis.com/projects/PROJECT_ID/locations/global/workloadIdentityPools/POOL_ID/providers/PROVIDER_ID"
+}
+```
+When sending to OIDC token, the endpoint is the `service_account_impersonation_url`, the header includes
+the `identity_token` from the previous step as a Bearer token, and the payload looks like:
+```json
+{
+  "delegates": [],
+  "scope": ["https://www.googleapis.com/auth/logging.write"],
+  "lifetime": "3600s"
+}
+```
+This call requests a GCP access token using the Identity Token
+
+### Risks and Mitigations
+
+This change involves configuration of authentication and is high-risk.
+The mitigation is to provide clear validation checks and logs for troubleshooting auth issues
```

Comment:
What specifically are we "mitigating"

Reply:
This is a template... not my words....   but we are mitigating the risks associated with modifying vector GCP authentication.